includes/abstracts/abstract-wc-order.php

```diff
@@ -1020,7 +1020,7 @@ protected function recalculate_coupons() {
 			$coupon_object = apply_filters( 'woocommerce_order_recalculate_coupons_coupon_object', $coupon_object, $coupon_code, $coupon_item, $this );
 
 			if ( $coupon_object ) {
-				$discounts->apply_coupon( $coupon_object );
+				$discounts->apply_coupon( $coupon_object, false );
```

Comment:
Can we add a comment in the code why we're skipping validation in this case?

```
Mike Jolley [12:31 PM] 
Ah no, this isn’t hacky. In this case, since the order already *has* the coupon, we need to skip re-validation
Otherwise things like usage limits will be reapplied which is wrong
```